tests/integration/test_charm.py

```diff
@@ -41,3 +26,19 @@ async def test_flask_is_up(flask_app, get_unit_ips):
         response = requests.get(f"http://{unit_ip}:8000", timeout=5)
         assert response.status_code == 200
         assert "Hello, World!" in response.text
+
+
+@pytest.mark.parametrize("update_config", [{"webserver_timeout": 7}], indirect=True)
+@pytest.mark.usefixtures("update_config")
+async def test_flask_webserver_config(
```

Comment:
Can we also come up with tests for the other configuration options?

Reply:
There isn't other configuration option right now, all configurations are web server configurations.

Reply:
Sorry, I misinterpreted your words before. The `webserver_timeout` is the only webserver configuration that's easy to probe from the server. Since the Flask can not access the Gunicorn configuration, this is the only way we have.

Reply:
@gregory-schiano could you look at this just to double check please?

Reply:
I think there could be a way using https://docs.gunicorn.org/en/latest/custom.html#custom-application

Doing something like: 
```python
from flask import Flask, jsonify
from gunicorn.app.base import BaseApplication

app = Flask(__name__)

class CustomGunicornApplication(BaseApplication):
    def __init__(self, app, options=None):
        self.options = options or {}
        self.application = app
        super().__init__()

    def load_config(self):
        config = {key: value for key, value in self.options.items()
                  if key in self.cfg.settings and value is not None}
        for key, value in config.items():
            self.cfg.set(key.lower(), value)

        # Store the Gunicorn config in the Flask app config
        self.application.config['gunicorn_config'] = config

    def load(self):
        return self.application

@app.route('/')
def hello():
    return 'Hello, World!'

@app.route('/config')
def config():
    gunicorn_config = app.config.get('gunicorn_config', {})
    return jsonify(gunicorn_config)

if __name__ == '__main__':
    options = {
        'bind': '%s:%s' % ('127.0.0.1', '8000'),
        'workers': 4,
    }
    CustomGunicornApplication(app, options).run()
```

Would probably require to inherit from: https://github.com/benoitc/gunicorn/blob/master/gunicorn/app/wsgiapp.py instead of BaseApplication

It sounds a bit complex, but should make it

Reply:
Indeed, that requires to change the command.
An alternative would be to add:
```python
def when_ready(server):
    app = server.app.callable

    # Extract Gunicorn configuration
    gunicorn_config = {key: value for key, value in server.cfg.settings.items()
                       if value.value is not None and value.value != ''}
    
    # Set Gunicorn configuration in the Flask app
    app.config['gunicorn_config'] = gunicorn_config
```

To the gunicorn config and then an endpoint like:
```python
@app.route('/config')
def config():
    gunicorn_config = app.config.get('gunicorn_config', {})
    return jsonify(gunicorn_config)
```

The question is now, how do we add this to the gunicorn configuration

Reply:
Config hook is very specific to Gunicorn and will not apply to other WSGI servers.

I believe the current implementation is sufficient to test the timeout configuration value. I have added some tests that use concurrent requests to test the threads and workers configuration values. I also repeated each experiment three times to increase the confidence level, or I can add more repetitions if you want.

I think the current integration tests are "scientific" enough.

Reply:
Sounds reasonable to me, we can always use the suggestion from Gregory if needed